lib/XTR/StatsMigrationTask.php

```diff
@@ -0,0 +1,481 @@
+
+            // Give SQL time to recover
+            if ($watermark > 0) {
+                $this->log->debug('Stats migration effected '.$count.' rows, sleeping.');
+                sleep($options['pauseBetweenLoops']);
+            }
+        }
+
+    }
+
+    public function moveStatsFromStatArchiveToStatMongoDb($options = [])
+    {
+
+        $options = array_merge([
+
+            'numberOfRecords' => 10000,
+            'numberOfLoops' => 1000,
+            'pauseBetweenLoops' => 10,
+        ], $options);
+
+        // Migration from stat table to Mongo
+        $this->migrationStatToMongo($options);
+
+        // Migration from stat_archive table to Mongo
+        // After migration delete only stat_archive
+        if ($this->archiveExist == true) {
+            $this->migrationStatArchiveToMongo($options);
+        }
+    }
+
+    function migrationStatToMongo($options)
+    {
+        // Stat Archive Task
+        $archiveTask = $this->taskFactory->getByClass('\Xibo\XTR\\StatsArchiveTask');
+
+        $fileName = $this->config->getSetting('LIBRARY_LOCATION') . '.watermark_stat_mongo.txt';
+
+        // Get low watermark from file
+        $watermark = $this->getWatermarkFromFile($fileName, 'stat');
+
+        $sql = $this->store->getConnection()->prepare('SELECT statId FROM stat WHERE statId < :watermark ORDER BY statId DESC LIMIT 1');
+        $sql->bindParam(':watermark', $watermark, \PDO::PARAM_INT);
+        $sql->execute();
+
+        // Mark the Stats Archiver as disabled if there are records in stat table and set option archiveStats off
+        if ($sql->rowCount() > 0) {
+
+            // Quit the StatsArchiveTask if it is running
+            if ($archiveTask->runNow == 1) {
+                $this->log->debug('Quitting the stat migration task as stat archive task is running');
+                return;
+            }
+            $archiveTask->isActive = 0;
+            $archiveTask->options['archiveStats'] = 'Off';
+            $archiveTask->save();
+            $this->store->commitIfNecessary();
+        }
+
+        $numberOfLoops = 0;
+        $count = 0;
+        while ($watermark > 0) {
+
+            $stats = $this->store->getConnection()
+                ->prepare('SELECT * FROM stat WHERE statId < :watermark ORDER BY statId DESC LIMIT :limit');
+            $stats->bindParam(':watermark', $watermark, \PDO::PARAM_INT);
+            $stats->bindParam(':limit', $options['numberOfRecords'], \PDO::PARAM_INT);
+
+            // Run the select
+            $stats->execute();
+
+            // Keep count how many stats we've inserted
+            $recordCount = $stats->rowCount();
+            $count+= $recordCount;
+
+            // End of records
+            if ($this->checkEndOfRecords($recordCount, $fileName) === true) {
+
+                // Enable the StatsArchiver task and set option archiveStats on
+                $archiveTask->isActive = 1;
+                $archiveTask->options['archiveStats'] = 'On';
```

Comment:
we shouldn't set the archiveStats option as the user might not want to archive stats - it should be the value it was when we started